packages-exp/auth-exp/src/model/action_code_url.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthErrorCode, AUTH_ERROR_FACTORY } from '../core/errors';
+import { Operation } from './action_code_info';
+import { Auth } from './auth';
+
+/**
+ * Enums for fields in URL query string.
+ * @enum {string}
+ */
+enum QueryField {
+  API_KEY = 'apiKey',
+  CODE = 'oobCode',
+  CONTINUE_URL = 'continueUrl',
+  LANGUAGE_CODE = 'languageCode',
+  MODE = 'mode',
+  TENANT_ID = 'tenantId'
+}
+
+/**
+ * Map from mode string in action code URL to Action Code Info operation.
+ */
+const _ModeToOperationMap: { [key: string]: Operation } = {
```

Comment:
So I _think_ we shouldn't be putting the `_` prefix on names that _aren't_ exported. These are all file-private. The `_` prefix is meant to denote exported (aka code public) names that shouldn't be used by other packages. See https://api-extractor.com/pages/tsdoc/tag_internal/ (Fei linked this to me in my PR)

Also file level consts should be UPPER_SNAKE_CASE